lib/experimental/auto-inserting-blocks.php

```diff
@@ -0,0 +1,246 @@
+<?php
+/**
+ * Auto-inserting blocks.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Return a function that auto-inserts blocks relative to a given block.
+ *
+ * @param array  $inserted_block    The block to insert.
+ * @param string $relative_position The position relative to the given block.
+ * @param string $anchor_block      The block to insert relative to.
```

Comment:
when I read this I assume it's taking a block object, but then I learn it's a block type. I think we can clarify here

```php
 * @param string $anchor_block_type Will auto insert given block when encountering this block type.
```